libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -437,7 +438,7 @@ public void checkFileWrite(Class<?> callerClass, Path path) {
     }
 
     public void checkCreateTempFile(Class<?> callerClass) {
-        checkFileWrite(callerClass, pathLookup.tempDir());
+        checkFileWrite(callerClass, pathLookup.getBaseDirPaths(TEMP).findFirst().get());
```

Comment:
Could we have a comment here explaining why this is "ok"? in production we only ever have 1 temp dir, and this is really a production check on entitlements.

Reply:
Added.